slasher/service/service.go

```diff
@@ -99,9 +103,19 @@ func (s *Service) Start() {
 	}).Info("Starting hash slinging slasher node")
 	s.context = context.Background()
 	s.startSlasher()
-	s.startBeaconClient()
-	go s.finalisedChangeUpdater()
+	if s.beaconClient == nil {
+		s.startBeaconClient()
```

Comment:
startBeaconClient looks like a function that could benefit from returning an error, as I expect something bad can happen in there that should prevent code below from running

Reply:
great, added